### ISSUE TITLE
[Suggestion] Report different weather

### ISSUE THREAD

**Issue by adem4ik** [FelixdelasPozas/TrayWeather#35]
Since the end of the spring TrayWeather shows me incorrect weather sometimes, and I usually report it via https://openweathermap.org/ site.
![image](https://user-images.githubusercontent.com/4707112/125738141-eae41cd2-396a-4002-a7b0-bfa7f2232e4b.png)

I think that would be quite useful to have a menu entry like «Report different weather» in TW. That entry should redirect to the weather provider site - https://openweathermap.org/
![image](https://user-images.githubusercontent.com/4707112/125738427-51f33eb2-1372-43ac-a296-1f7e34987dcb.png)

**Comment by FelixdelasPozas** [comment]
Hi.

> Since the end of the spring TrayWeather shows me incorrect weather sometimes (...)

If the weather is incorrect or just different to what the web shows its because a miss in geo-location. OpenWeatherMap uses latitude and longitude in the data request and if the geo-location is incorrect it will show the weather from an incorrect location. I added the "Use DNS" option in the configuration dialog because one time the geo-location was reporting that I was in another city (about 400 km away) maybe because my ISP was re-routing all traffic and giving IPs from that other city. Using the DNS IP for geo-location may correct that. 

Another option is to locate where you are (in maps.google.es for example, using "show my location") and get the longitude and latitude (appears in the web address after you use "show my location") and use that information in the configuration dialog (and not use automatic geo-location). OpenWeatherMap should return the weather data from the closest weather station.

In the screenshot you show I get that TrayWeather and OpenWeatherMap are reporting 23º, but the correct temperature is 27º? Is that correct?

Try the "Use DNS" option or use the longitude and latitude from [Google maps](maps.google.es). 

> I think that would be quite useful to have a menu entry like «Report different weather» in TW. That entry should redirect to the weather provider site - https://openweathermap.org/

About adding a link to OpenWeatherMap, that would be easy but I think that will invalidate the purpose of the application, as the reason I made TrayWeather is precisely to have that weather information and forecast one-click-away instead of going to the web. If TrayWeather is correctly configured and the geo-location reports your location accurately you shouldn't need to do it. 

I will leave the issue open if someone wants to discuss it, but for now I think a link to the weather provider is not necessary, only in the "About" and "Configuration" dialogs, but not in the menu. 

Regards,
Félix.

**Comment by adem4ik** [comment]
I don't think it was a miss in geolocation. As screenshot shows both TrayWeather and OpenWeather had the same temperature value - 23º. But other weather provider (actually many of them) had different value - about 27º. So the problem comes from OpenWeather data, not from TrayWeather configuration nor anything alse. And that's why I suggest such feature.

**Comment by FelixdelasPozas** [comment]
Recently I've seen [this](https://github.com/public-apis/public-apis#weather), a list of public weather api. Maybe with a little time I can implement an alternative to OpenWeatherMap, also free. I must look at it. If the failure is in OpenWeatherMap data, that is something I cannot correct. 

Regards, 
Félix.

**Comment by FelixdelasPozas** [comment]
I am going to close this because from version 1.31 there is the possibility of using different weather providers. The list of possible alternatives to implement is in the [wiki](https://github.com/FelixdelasPozas/TrayWeather/wiki/Providers).
